updater/versions/update-8.py

```diff
@@ -0,0 +1,195 @@
+@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@o:::::::o@c:::::::O@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+@@@@@@@@@@@@@@@@@@@@@@@@@@@@@8::::c:::C@:::c::::c@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+@@@@@@@@@@@@@@@@@@@@@@@@@@@@@C::::::c@@@@8:::::::@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@OCO@@@@@@@@@@@@8CC8@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@[0m
+[36m@@Co@@@@@@@@@@@@@@@@@@@@@@@@@:O@@@@@@O:@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+@@oc@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@O.@@@@@@@@@@@@@@@@@@@@oc@@@@@@@@@@@@@@@@@@@
+@@oc@@@@@@@O:coo:C@@:ccoo:o@@.O@O:coo::@@O.ooo:O@@:::oc:c@o::c8@c:co::O@c::::8@@
+@@oc@@@@@@8.@@@@@co@:C@@@@:C@:OO.@@@@O.@C:@@@@@cC@cc@@@@:c@oc@@::@@@@8:8C:O@@@@@
+@@oc@@@@@@O:888888@@:C@@@@co@:OC:@@@@O.@oc@@@@@co@:c@@@@cc@cc@@::OOOOOO@C:8@@@@@
+@@o:888888@cc@@@Cc@@::O@@C.@@:O@:o@@O::@@:o@@@o:@@c:o@@C:C@o:@@O:o@@8co@C:8@@@@@
+@@@@@@@@@@@@@@88@@@@:C@88@@@@@@@@@88@@@@@@@@8@@@@@cc@OO8@@@@8C8@@@OCO@@@@8@@@@@@
+@@@@@@@@@@@@@@@@@@@@:C@@@@@@@@@@@@@@@@@@@@@@@@@@@@:c@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@[0m
+"""
+
+BOOT_CONFIG = """
+# Minimal (with no vc_vchi_sm_init failure) GPU memory for headless mode
+gpu_mem=32
+# Enable the hardware watchdog
+dtparam=watchdog=on
+# Turn power LED into heartbeat
+dtparam=pwr_led_trigger=heartbeat
+# Enable the hardware random number generator (RNG)
+dtparam=random=on
+"""
+WATCHDOG_CONF_PATH = "/etc/watchdog.conf"
+SUDOERS_CONF_PATH = "/etc/sudoers"
+MOTD_HEAD_PATH = "/etc/motd.head"
+BOOT_CONFIG_PATH = "/boot/config.txt"
+
+def write_watchdog_conf():
+    with open(WATCHDOG_CONF_PATH, "w") as out_file:
+        out_file.write(WATCHDOG_CONF)
+
+def write_sudoers_conf():
+    with open(SUDOERS_CONF_PATH, "w") as out_file:
+        out_file.write(SUDOERS_CONF)
+
+def write_motd_head():
+    with open(MOTD_HEAD_PATH, "w") as out_file:
+        out_file.write(MOTD_HEAD)
+
+def write_boot_config():
+    with open(BOOT_CONFIG_PATH, "w") as out_file:
+        out_file.write(BOOT_CONFIG)
+
+def _perform_update():
+    check_call(["apt-get", "-q", "update"])
+    # Install OpenVPN package without starting the OpenVPN service
+    check_call(["apt-get", "-y", "install", "openvpn"], env={"RUNLEVEL": "1"})
+    # Install newest e2fsprogs due to incompatibly with ext4 file system checks
+    check_call(["apt-get", "-y", "install", "-t", "jessie-backports", "e2fsprogs"])
+    # Install obfs4proxy (includes a lite version of meek)
+    check_call(["apt-get", "-y", "install", "-t", "stretch", "obfs4proxy"])
+    # Remove old system versions of python-openssl
+    check_call(["apt-get", "-y", "remove", "python-openssl"])
+
+    # Remove bcm2708_rng entry as modules are now handled by /boot/config.txt
+    check_call(["cp", "-b", "/etc/modules", "/etc/modules.bak"])
+    check_call(["sed", "-i",  "'/bcm2708_rng/d'", "/etc/modules"])
+
+    # Disable OpenVPN systemd unit
+    check_call(["systemctl", "disable", "openvpn"])
```

Comment:
> If they are not started what is the purpose of disabling the openvpn service? 

The daemons are not started during apt installation. If the systemd unit of openvpn is not disabled it will be started on the next boot.

> I thought the point of this command was that of disabling the openvpn daemon and therefore I was thinking that it would be better to do that as soon as it's installed rather than later.

I find it a bit cleaner to have all packages [installed/unistalled](https://github.com/OpenObservatory/lepidopter-update/blob/9760262a8369d295a1413f8bb9acca887e9130e8/updater/versions/update-8.py#L145-L153)  first and then proceed with the rest of the update operations.

Whatever it works better to release the update ASAP as some lepidopter users are being affected by: https://github.com/meejah/txtorcon/issues/227